### PR TITLE
fix(devices): Avoid spurious writes to device record if nothing has changed

### PR DESCRIPTION
**Pull request by rfk**
@jrgm what do you think of something like this?

### CODE REVIEW

**Comment by vladikoff** [comment]
yesssss!

**Comment by rfk** [comment]
Updated to `log.info({ op: 'Account.device.spuriousUpdate' })` when we skip a write, for monitoring purposes.  Also because I enjoy using the word "spurious".

**Comment by rfk** [comment]
> yesssss!

@vladikoff was that an r+? ;-)

File: lib/routes/account.js [hunk 1]
```diff
@@ -840,6 +840,17 @@ module.exports = function (
         log.begin('Account.device', request)
         var payload = request.payload
         var sessionToken = request.auth.credentials
+        // Clients have been known to send spurious device updates,
+        // which generates lots of unnecessary database load.
+        // Don't write out the update if nothing has actually changed.
+        if (payload.id && payload.id === sessionToken.deviceId.toString('hex') &&
```
Comment (vladikoff) [inline lib/routes/account.js]: @rfk in https://github.com/mozilla/fxa-auth-server/blob/master/docs/api.md#post-v1accountdevice 

`If no device id is specified, both name and type must be provided. ` - will this fail when `id` is missing?

**Comment by jrgm** [comment]
LGTM, but not really reviewing the patch for correctness, since I can't.

File: lib/routes/account.js (hunk 1 above)
Comment (vladikoff) [inline lib/routes/account.js]: Or do we not need to worry about that case for this ?

Comment (rfk) [inline lib/routes/account.js]: If the id is missing then it's a new device registration, the condition should fail and it should fall through to the existing code.

Comment (rfk) [inline lib/routes/account.js]: to be clear: if it's missing the id, then we want to always process the write rather than skipping it

Comment (vladikoff) [inline lib/routes/account.js]: > to be clear: if it's missing the id, then we want to always process the write rather than skipping it 

👍

**Comment by vladikoff** [comment]
@rfk looks good, we might need to pull request this against an older auth-server or is `master` fine?

**Comment by rfk** [comment]
thanks @vladikoffl; yeah I'll backport to release branch as well for deployment

**Pull request merged**